community/csv/src/main/java/org/neo4j/csv/reader/MultiReadable.java

```diff
@@ -46,55 +47,59 @@ public MultiReadable( RawIterator<CharReadable,IOException> actual ) throws IOEx
     }
 
     @Override
-    public int read( char[] buffer, int offset, int length ) throws IOException
+    public SectionedCharBuffer read( SectionedCharBuffer buffer, int from ) throws IOException
     {
-        int read = 0;
-        while ( read < length )
+        buffer.compact( buffer, from );
+        while ( current != null )
         {
-            int readThisTime = current.read( buffer, offset + read, length - read );
-            if ( readThisTime == -1 )
+            // Actually, we sort of assume that the returned buffer is the same as the one we pass in here
+            buffer.readFrom( current );
```

Comment:
Oh wait, that was from a previous idea of how this should work. I'll remove the comment since it's no longer valid.